core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -321,8 +321,11 @@ public T get(Settings settings) {
     public String getRaw(Settings settings) {
         // They're using the setting, so we need to tell them to stop
         if (this.isDeprecated() && this.exists(settings)) {
+            if (deprecationLogger.get() == null) {
```

Comment:
No, you're right it's thread unsafe however we could get bit by it. Since it does not really matter if this field is set twice, I'll just push a commit that removes the usage of `SetOnce`. Thanks for noticing.